tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/ConstSimplifierBase.scala

```diff
@@ -29,49 +29,89 @@ abstract class ConstSimplifierBase {
 
     // integer operations
     case OperEx(TlaArithOper.plus, ValEx(TlaInt(left)), ValEx(TlaInt(right))) =>
-      if (left == 0) {
-        ValEx(TlaInt(right))(intTag)
-      } else if (right == 0) {
-        ValEx(TlaInt(left))(intTag)
+      ValEx(TlaInt(left + right))(intTag)
+
+    // 0 + x = x
+    case OperEx(TlaArithOper.plus, ValEx(TlaInt(left)), rightEx) if left == 0 => rightEx
+    // x + 0 = x
+    case OperEx(TlaArithOper.plus, leftEx, ValEx(TlaInt(right))) if right == 0 => leftEx
+
+    // Evaluate constant subtraction
+    case OperEx(TlaArithOper.minus, ValEx(TlaInt(left)), ValEx(TlaInt(right))) => ValEx(TlaInt(left - right))(intTag)
+    // 0 - x = -x
+    case OperEx(TlaArithOper.minus, ValEx(TlaInt(left)), rightEx) if left == 0 =>
+      OperEx(TlaArithOper.uminus, rightEx)(intTag)
+    // x - 0 = x
+    case OperEx(TlaArithOper.minus, leftEx, ValEx(TlaInt(right))) if right == 0 => leftEx
```

Comment:
Any particular reason you've opted for guards here (the `if .... => `) rather than just matching against the `0` value?

E.g.,

```suggestion
    case OperEx(TlaArithOper.plus, ValEx(TlaInt(0)), rightEx) => rightEx
    // x + 0 = x
    case OperEx(TlaArithOper.plus, leftEx, ValEx(TlaInt(0))) => leftEx

    // Evaluate constant subtraction
    case OperEx(TlaArithOper.minus, ValEx(TlaInt(left)), ValEx(TlaInt(right))) => ValEx(TlaInt(left - right))(intTag)
    // 0 - x = -x
    case OperEx(TlaArithOper.minus, ValEx(TlaInt(0)), rightEx) => OperEx(TlaArithOper.uminus, rightEx)(intTag)
    // x - 0 = x
    case OperEx(TlaArithOper.minus, leftEx, ValEx(TlaInt(0))) => leftEx
```

Reply:
Yes! I actually explained it here: https://github.com/informalsystems/apalache/pull/1191#discussion_r774526282

Reply:
I see! My bad. Makes perfect sense.